src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs

```diff
@@ -153,6 +154,14 @@ public static class WebScriptHostBuilderExtension
                     services.TryAddEnumerable(ServiceDescriptor.Singleton<IHostedService, IFileMonitoringService>(p => p.GetService<IFileMonitoringService>()));
 
                     ConfigureRegisteredBuilders(services, rootServiceProvider);
+
+                    // Hidrate ExtensionOptionProvider to FunctionSyncManager
```

Comment:
Not sure why all this is needed. Can't you simply add the new service as I did [here](https://github.com/Azure/azure-functions-host/blob/d7ac580aad39489592f3c64d88d52fb4517d5593/src/WebJobs.Script.WebHost/WebScriptHostBuilderExtension.cs#L157) and let DI take care of everything else?

Reply:
I explained at the https://github.com/Azure/azure-functions-host/pull/7863#discussion_r794094949